src/coreclr/src/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.cs

```diff
@@ -0,0 +1,224 @@
+
+            return retValue!;
+        }
+
+        [DllImport(RuntimeHelpers.QCall)]
+        private static extern void GetOrCreateObjectForComInstanceInternal(ObjectHandleOnStack comWrappersImpl, IntPtr externalComObject, CreateObjectFlags flags, ObjectHandleOnStack retValue);
+
+        /// <summary>
+        /// Create a managed object for the object pointed at by <paramref name="agileObjectRef"/> respecting the values of <paramref name="flags"/>.
+        /// </summary>
+        /// <param name="externalComObject">Object to import for usage into the .NET runtime.</param>
+        /// <param name="agileObjectRef">IAgileReference pointing at the object to import into the .NET runtime.</param>
+        /// <param name="flags">Flags used to describe the external object.</param>
+        /// <returns>Returns a managed object associated with the supplied external COM object.</returns>
+        /// <remarks>
+        /// The <paramref name="agileObjectRef"/> is an <see href="https://docs.microsoft.com/windows/win32/api/objidl/nn-objidl-iagilereference">IAgileReference</see> instance. This type should be used to ensure the associated external object, if not known to be free-threaded, is released from the correct COM apartment.
+        /// </remarks>
+        protected abstract object CreateObject(IntPtr externalComObject, IntPtr agileObjectRef, CreateObjectFlags flags);
+
+        // Call to execute the abstract instance function
+        internal static object CallCreateObject(ComWrappers? comWrappersImpl, IntPtr externalComObject, IntPtr agileObjectRef, CreateObjectFlags flags)
+            => (comWrappersImpl ?? s_globalInstance!).CreateObject(externalComObject, agileObjectRef, flags);
+
+        /// <summary>
+        /// Called when a request is made for a collection of objects to be released.
+        /// </summary>
+        /// <param name="objects">Collection of objects to release.</param>
+        /// <remarks>
+        /// The default implementation of this function throws <see cref="System.NotImplementedException"/>.
+        /// </remarks>
+        protected virtual void ReleaseObjects(IEnumerable objects)
+        {
+            throw new NotImplementedException();
+        }
+
+        // Call to execute the virtual instance function
+        internal static void CallReleaseObjects(ComWrappers? comWrappersImpl, IEnumerable objects)
+            => (comWrappersImpl ?? s_globalInstance!).ReleaseObjects(objects);
+
+        /// <summary>
+        /// Register this class's implementation to be used when a Reference Tracker Host instance is requested from another runtime.
+        /// </summary>
+        /// <remarks>
+        /// This function can only be called a single time. Subsequent calls to this function will result
+        /// in a <see cref="System.InvalidOperationException"/> being thrown.
+        /// </remarks>
+        public void RegisterForReferenceTrackerHost()
```

Comment:
I am wondering whether it would make sense to call this `RegisterAsGlobalInstance` - that is easier to understand name, and extend its meaning beyond reference tracking: The global instance can be also used to light-up the existing `Marshal` APIs.